app/controllers/idv/capture_doc_status_controller.rb

```diff
@@ -16,6 +16,8 @@ def document_capture_session_poll_render_result
       document_capture_session = DocumentCaptureSession.find_by(uuid: session_uuid)
       return { plain: 'Unauthorized', status: :unauthorized } unless document_capture_session
 
+      puts "POLL: #{document_capture_session.result_id} || #{flow_session[:document_capture_session_uuid]}"
```

Comment:
cc @stevegsa @mitchellhenke @aduth 

After fixing some missing URLs (the `capture_doc_flow` sections) I have discovered that place we put the data for mobile and the place we read data from back on the desktop are two separate places. The output of the puts statements:

```
VERIFY: a924bd75-e186-46db-bf9b-c37694d77854
POLL:  || 75c6bfc4-a610-4c2d-86d8-7d57183c27be
```

So this has to do with how we split up the multiple async result keys, `document_capture_session_uuid_key` vs `verify_document_capture_session_uuid_key`

I'm not seeing a clear path to unifying, will take a look tomorrow with fresher eyes

Reply:
> So this has to do with how we split up the multiple async result keys, `document_capture_session_uuid_key` vs `verify_document_capture_session_uuid_key`

I'm not totally familiar with how the async flow sets up its own session UUID, but... if it's challenging to unify, could it be assumed that if the verify (async) session key exists, it should be preferred, and check both?

i.e.

```
session_uuid = flow_session[verify_document_capture_session_uuid_key] || flow_session[document_capture_session_uuid_key]
```

Not sure if we need to cover all these variants also:

https://github.com/18F/identity-idp/blob/e0c23c99803512b0c7a9765b41e981874c018744/app/services/idv/steps/doc_auth_base_step.rb#L117-L135

Reply:
So originally there was only one key, `document_capture_session_uuid_key`, since the only asynchronous action that required polling was the desktop polling to wait for the mobile flow to finish.

Since now we have 3 asynchronous steps, each one of those got their own key, and then the separate flows (cac, recover, etc) got their own key hence all the keys.

Ideally I think we could remove one, or link them to each other better, but it needs some noodling.

I think the fastest fix/patch would definitely be to just check two keys in this controller...I don't like it because to me that feels like not doing the work to fully explore all the paths, but at least it would function.

Reply:
Made some patches and got it working, dropped comments at the various points